### PR TITLE
Fix `HttpServerFactory` to enable coroutine().

### PR DESCRIPTION
**Pull request by Moln**
- Fix `HttpServerFactory` to enable coroutine().
- Fix `SwooleStreamTest`,  change `rawcontent` to `rawContent` (To camel case).
- ~Add `swoole/ide-helper` in `composer.json` .~

Provide a narrative description of what you are trying to accomplish:

- [x] Are you fixing a bug?
  - [x] Detail how the bug is invoked currently.
  - [x] Detail the original, incorrect behavior.
  - [x] Detail the new, expected behavior.
  - [x] Base your feature on the `master` branch, and submit against that branch.
  - [x] Add a regression test that demonstrates the bug, and proves the fix.
  - [ ] Add a `CHANGELOG.md` entry for the fix.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "require-dev": {
         "filp/whoops": "^2.1",
         "phpunit/phpunit": "^7.1.1",
+        "swoole/ide-helper": "dev-master",
```
Comment (weierophinney) [inline composer.json]: What does this provide, exactly? Is it just stubs for autocompletion?

If so, may I request that you open a separate PR for just this change, please?

File: src/HttpServerFactory.php [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 use Psr\Container\ContainerInterface;
 use Swoole\Http\Server as SwooleHttpServer;
-use SwooleRuntime;
+use Swoole\Runtime as SwooleRuntime;
```
Comment (weierophinney) [inline src/HttpServerFactory.php]: Is this true across all versions of Swoole?

File: test/SwooleStreamTest.php [hunk 3]
```diff
@@ -26,7 +26,7 @@ public function setUp()
         }
         $this->request = $this->prophesize(SwooleHttpRequest::class);
         $this->request
-            ->rawcontent()
+            ->rawContent()
```
Comment (weierophinney) [inline test/SwooleStreamTest.php]: Is this true across all Swoole versions? We grabbed this name directly from source at one point, which is why I'm asking...

Comment (Moln) [reply test/SwooleStreamTest.php]: https://github.com/swoole/swoole-src/commit/b8b4c3917b4a9bc61851a9547506d3bc444b6905#diff-3ced702c2439aa81b4faccece28f4c90

It's looks no changelog in [swoole@v4.3.0](https://github.com/swoole/swoole-src/releases/tag/v4.3.0).

C extension can be ignore case，when calls the `$swooleRequest->rawContent()`.

File: src/HttpServerFactory.php (hunk 2 above)
Comment (Moln) [reply src/HttpServerFactory.php]: Yes, I saw all the changes in the [`swoole_runtime.cc`](https://github.com/swoole/swoole-src/commits/master/swoole_runtime.cc) file.

File: test/SwooleStreamTest.php (hunk 3 above)
Comment (huangzhhui) [reply test/SwooleStreamTest.php]: @weierophinney Method name is not case sensitive, `rawContent()` would be a better name.

**Comment by Moln** [comment]
@weierophinney  Can you clear cache and restart build of travis-ci ?

"The Travis CI build failed", It looks like CI cache problems.

**Comment by weierophinney** [comment]
> @weierophinney Can you clear cache and restart build of travis-ci ?
> 
> "The Travis CI build failed", It looks like CI cache problems.

Done, but I'm still seeing errors...

**Comment by Moln** [comment]
> > @weierophinney Can you clear cache and restart build of travis-ci ?
> > "The Travis CI build failed", It looks like CI cache problems.
> 
> Done, but I'm still seeing errors...

Fixed!

**Comment by webimpress** [comment]
@Moln Can you describe what is your fix?
I've pulled your test to my local instance any without change in `src/HttpServerFactory.php` test is green as well. I am not sure what is the issue and how this PR fixes it?

File: test/HttpServerFactoryTest.php [hunk 4]
```diff
@@ -294,4 +296,38 @@ public function testServerCanBeStartedForKnownSocketTypeCombinations($socketType
...
+        $this->container->get('config')->willReturn([
+            'zend-expressive-swoole' => [
+                'enable_coroutine' => true,
+            ],
+        ]);
+        $factory = new HttpServerFactory();
+        $factory($this->container->reveal());
+
+        if (! method_exists(SwooleRuntime::class, 'enableCoroutine')) {
+            return ;
```
Comment (webimpress) [inline test/HttpServerFactoryTest.php]: I believe this is due to Swoole prev to v4, right?
We shouldn't just return, but test should be marked as skipped in that case.
I believe it should be at the beginning of the test.

Comment (weierophinney) [reply test/HttpServerFactoryTest.php]: I agree with @webimpress here. Let's get that in place, and I'll do final review.

Thanks, @Moln

Comment (weierophinney) [reply test/HttpServerFactoryTest.php]: I've pushed this change to your branch.

Comment (Moln) [reply test/HttpServerFactoryTest.php]: 👍 👍 👍 

Sorry, I'm too busy at work. And thanks for your time to review and change this branch.

**Pull request merged**